src/core/rubberbandmodel.cpp

```diff
@@ -193,6 +196,9 @@ QgsPoint RubberbandModel::currentPoint( const QgsCoordinateReferenceSystem &crs,
 
 QgsPoint RubberbandModel::currentCoordinate() const
 {
+  if ( mPointList.isEmpty() )
+    return QgsPoint();
+
   return mPointList.at( mCurrentCoordinateIndex );
```

Comment:
```suggestion
  return mPointList.value( mCurrentCoordinateIndex );
```

that should protect for any index out of bounds.